lib/portlayer/exec/exec.go

```diff
@@ -113,11 +113,14 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 func eventCallback(ie events.Event) {
 	// grab the container from the cache
 	container := Containers.Container(ie.Reference())
+
 	if container != nil {
+		container.m.Lock()
```

Comment:
`swapState` already has a lock in there for the map, and you are using the atomic functions to change the state, so given that, is the lock necessary?